loader/src/main/java/net/minecraftforge/fml/loading/moddiscovery/ModValidator.java

```diff
@@ -72,7 +73,18 @@ public ITransformationService.Resource getPluginResources() {
     }
 
     public ITransformationService.Resource getModResources() {
-        return new ITransformationService.Resource(IModuleLayerManager.Layer.GAME, this.candidateMods.stream().map(IModFile::getSecureJar).toList());
+        Stream<ModFile> modFileStream = this.candidateMods.stream();
+        if (FMLEnvironment.production) {
+            // In production, only allow game libraries and/or mods in the loading mod list
+            // This prevents custom mixins from loading if there is a dependency error
+            // Usually, these mixins will break due to a missing class or AT, and that
+            // will prevent our error screen from ever becoming visible.
+            Set<IModFile> validMods = new HashSet<>();
+            validMods.addAll(this.loadingModList.getModFiles().stream().map(ModFileInfo::getFile).toList());
+            validMods.addAll(this.gameLibraries);
+            modFileStream = modFileStream.filter(validMods::contains);
```

Comment:
Why not `validMods.stream()`?

Reply:
Sure, same thing in the end.

Reply:
FML code is already hard enough to read as is... :P